llvm/lib/Support/NativeFormatting.cpp

```diff
@@ -262,5 +262,17 @@ size_t llvm::getDefaultPrecision(FloatStyle Style) {
   case FloatStyle::Percent:
     return 2; // Number of decimal places.
   }
+
+  // Workaround for MSVC bug in VS2022:
+  // https://developercommunity.visualstudio.com/t/Prev-Issue---with-__assume-isnan-/1597317
+  // llvm_unreachable expands to __assume(false) with MSVC which triggers the
+  // bug.
+#if _MSC_VER >= 1930
```

Comment:
```suggestion
#if _MSC_VER >= 1930 && _MSC_VER <= 1932
```
this was fixed in 17.3 (1933).